sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -56,6 +56,71 @@ public class BlobBaseClient
         /// </summary>
         public virtual CustomerProvidedKey? CustomerProvidedKey => _customerProvidedKey;
 
+        /// <summary>
+        /// The Storage account name corresponding to the blob client.
+        /// </summary>
+        private string _accountName;
+
+        /// <summary>
+        /// Gets the Storage account name corresponding to the blob client.
+        /// </summary>
+        public string AccountName
+        {
+            get
+            {
+                SetNameFieldsIfNull();
+                return _accountName;
+            }
+        }
+
+        /// <summary>
+        /// The container name corresponding to the blob client.
+        /// </summary>
+        private string _containerName;
+
+        /// <summary>
+        /// Gets the container name corresponding to the blob client.
+        /// </summary>
+        public string ContainerName
+        {
+            get
+            {
+                SetNameFieldsIfNull();
+                return _containerName;
+            }
+        }
+
+        /// <summary>
+        /// The name of the blob.
+        /// </summary>
+        private string _name;
+
+        /// <summary>
+        /// Gets the name of the blob.
+        /// </summary>
+        public string Name
+        {
+            get
+            {
+                SetNameFieldsIfNull();
+                return _name;
+            }
+        }
+
+        /// <summary>
+        /// Sets the various name fields if they are currently null.
+        /// </summary>
+        private void SetNameFieldsIfNull()
```

Comment:
Can you move this below the .ctors?  The general structure of our files is Fields/Properties, ctors, helpers (like this), and finally service methods.